scripts/lib/CIME/case/case_run.py

```diff
@@ -106,29 +115,36 @@ def _run_model_impl(case, lid, skip_pnl=False, da_cycle=0):
         stat = run_and_log_case_status(run_func, "model execution", caseroot=case.get_value("CASEROOT"))
         model_logfile = os.path.join(rundir, model + ".log." + lid)
         # Determine if failure was due to a failed node, if so, try to restart
-        if node_fail_re:
-            node_fail_regex = re.compile(node_fail_re)
+        if retry_run_re or node_fail_re:
             model_logfile = os.path.join(rundir, model + ".log." + lid)
             if os.path.exists(model_logfile):
-                num_fails = len(node_fail_regex.findall(open(model_logfile, 'r').read()))
-                if num_fails > 0 and case.spare_nodes >= num_fails:
+                num_node_fails=0
+                num_retry_fails=0
+                if node_fail_re:
+                    num_node_fails = len(node_fail_regex.findall(open(model_logfile, 'r').read()))
+                if retry_run_re:
+                    num_retry_fails = len(retry_run_regex.findall(open(model_logfile, 'r').read()))
+                logger.debug ("RETRY: num_retry_fails {} spare_nodes {} retry_count {}".
+                              format(num_retry_fails, case.spare_nodes, retry_count))
+                if num_node_fails > 0 and case.spare_nodes >= num_node_fails:
                         # We failed due to node failure!
                     logger.warning("Detected model run failed due to node failure, restarting")
-
+                    case.spare_nodes -= num_node_fails
+                    loop = True
+                elif num_retry_fails > 0 and retry_count >= num_retry_fails:
```

Comment:
Do you want mpirun-retry attempts to decrease by num_retry_fails or 1? I would think 1. The reason we decrease by num_node_fails above is because we assume multiple occurrences of the regex means multiple nodes failed. I wouldn't think you'd want mpi retry to work that way, since it could be printing the regex message NP times.

Reply:
In the current cheyenne case num_retry_fails will always be 1.   I could change this in anticipation of some other issue that is reported NP times or the user could just be clever with the REGEX.

Reply:
Is there any case where we wouldn't want it to be 1?

Reply:
I can't think of a case - I will make this change.